server/component-api/check-dependencies.js

```diff
@@ -6,101 +6,99 @@ const Path = require("path");
+const ensureDirectoryExists = require("../utils/ensure-directory-exists");
 
 const prefixes = Config.MODULE_PREFIXES_INCLUDE;
 const pattern = prefixes && prefixes.length && new RegExp(prefixes.join("|"));
 
 const getDepLatest = (dep, version, isDev) => {
   return new Promise((resolve, reject) => {
+    execFile(
+      "bash",
+      [Path.join(__dirname, "../../scripts/info-module.sh"), dep],
+      (err, stdout, stderr) => {
+        if (err) {
+          console.log("error getting module info", err);
+          return reject(err);
+        }
 
-    execFile("bash", [Path.join(__dirname, "../../scripts/info-module.sh"), dep], (err, stdout, stderr) => {
-      if (err || stderr) {
-        console.log("error getting module info", err || stderr);
-        return resolve({});
-      }
+        if (stderr) {
+          console.error("module info stderr", stderr);
+          return reject(new Error(stderr));
+        }
 
-      const m = stdout.match(/latest:\s'([\d\.]+)'/);
-      let wantedVersion = m ? m[1] : version.replace(/[^\.\d]/g, "");
+        const m = stdout.match(/latest:\s'([\d\.]+)'/);
+        let wantedVersion = m ? m[1] : version.replace(/[^\.\d]/g, "");
 
-      if (!(/\./).test(wantedVersion)) {
-        wantedVersion += ".0.0";
-      }
+        if (!/\./.test(wantedVersion)) {
+          wantedVersion += ".0.0";
+        }
 
-      resolve({
-        uri: "#",
-        displayName: dep,
-        version: checkVersion(wantedVersion, version),
-        description: isDev ? "[dev]" : ""
-      });
-    });
+        resolve({
+          uri: "#",
+          displayName: dep,
+          version: checkVersion(wantedVersion, version),
+          description: isDev ? "[dev]" : ""
+        });
+      }
+    );
   });
 };
 
 const processDeps = (deps, isDev) => {
-
   const promises = [];
 
   if (deps) {
-    Object.keys(deps).map((dep) => {
-
-      if (pattern && !(pattern).test(dep)) {
+    Object.keys(deps).map(dep => {
+      if (pattern && !pattern.test(dep)) {
         return;
       }
 
       const promise = getDepLatest(dep, deps[dep], isDev);
       promises.push(promise);
-
     });
   }
 
   return Promise.all(promises);
-
 };
 
 const checkDepVersions = (deps, isDev) => {
-  return processDeps(deps, isDev)
-    .then((depArr) => {
-      return depArr;
-    });
+  return processDeps(deps, isDev).then(depArr => {
+    return depArr;
+  });
 };
 
-const writeDeps = (moduleName, moduleDeps) => {
-
+const writeDeps = (moduleName, meta, moduleDeps) => {
   if (!moduleDeps.length) {
     return;
   }
 
-  return new Promise((resolve) => {
-    Fs.readFile(Path.join(__dirname, `../../data/${moduleName}.json`), (err, repoFile) => {
-      let data = {};
-      try {
-        data = JSON.parse(repoFile);
-      } catch (e) {}
-
-      data.deps = moduleDeps;
+  // Fetch the repo data file if it already exists
+  let repoFile = {};
+  let writePath = Path.join(__dirname, `../../data/${moduleName}.json`);
```

Comment:
Shall we use `const` here since writePath won't change.

And if try catch is to verify if the path exists, you can use optional-require: https://www.npmjs.com/package/optional-require